joern-cli/frontends/csharpsrc2cpg/src/test/scala/io/joern/csharpsrc2cpg/querying/ast/ControlStructureTests.scala

```diff
@@ -25,7 +25,7 @@ class ControlStructureTests extends CSharpCode2CpgFixture {
             case Some(exp: Call) =>
               exp.code shouldBe "new Exception(\"Error!\")"
               exp.name shouldBe Defines.ConstructorMethodName
-              exp.typeFullName shouldBe "Exception" // TODO: Should really be System.Exception
+              exp.typeFullName shouldBe "System.InsufficientMemoryException"
```

Comment:
This doesn't seem valid?

Reply:
https://github.com/Privado-Inc/joern/blob/35698e1b283485fb65cb0f68afa5244f64a85111/joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/datastructures/ProgramSummary.scala#L94

This line picks up all the types in scope, which end with the typeName parameter. What is the best way to handle this?

Reply:
Maybe, we can match on the last part ( part meaning types split on "." ) ?

Reply:
@karan-batavia yes let's do that, match the last elements in each respective split